pkg/kubelet/dockertools/manager.go

```diff
@@ -1617,7 +1618,21 @@ func (dm *DockerManager) computePodContainerChanges(pod *api.Pod, runningPod kub
 				// RestartPolicy::Always, but it's not a big deal.
 				glog.V(3).Infof("Container %+v is dead, but RestartPolicy says that we should restart it.", container)
 				containersToStart[index] = empty{}
+
+			}
+
+			// Trying to get the reference for container,
+			// if failing to get the ref giv the Pod as reference
+			var containerOrPodRef runtime.Object
+			containerOrPodRef = pod // default
+			ref, err := kubecontainer.GenerateContainerRef(pod, &container)
+			if ref != nil && err == nil {
+				containerOrPodRef = ref
 			}
+
+			// sending the event "died" weither or not the container should be restarted
+			dm.recorder.Eventf(containerOrPodRef, "Died", "Container \"%v\" in pod \"%v\" died.", container.Name, podFullName)
```

Comment:
@yujuhong I notice the event sent on pod creation. And I was looking for a way to avoid this. 

Regarding the « restart policy == never » case, I did not figure this out.

Can you propose an algorithm thad manage all the cases ?